storm-server/src/main/java/org/apache/storm/scheduler/ISchedulingState.java

```diff
@@ -339,4 +354,31 @@ boolean wouldFit(
      * Get the nimbus configuration.
      */
     Map<String, Object> getConf();
+
+    /**
+     * Determine the list of racks on which topologyIds have been assigned. Note that the returned set
+     * may contain {@link DNSToSwitchMapping#DEFAULT_RACK} if {@link #getHostToRack()} is null or
+     * does not contain the assigned host.
+     *
+     * @param topologyIds for which assignments are examined.
+     * @return set of racks on which assignments have been made.
+     */
+    default Set<String> getAssignedRacks(String... topologyIds) {
```

Comment:
This will be used in defragmentation code.

Reply:
Does it make sense to delete this from this PR and move it to the PR for defragmentation code? It makes it easier to understand when looking back the commit history in the future

Reply:
This method gets used in testing for the scheduling. And the right place (based on the interface) seems to be here. Why will developers have to worry, they don't have to implement this method.
https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html

Reply:
It makes the interface unnecessary bigger since this method is only used in test cases. I understand devs don't need to implement this method. But they need to maintain the code. My personal opinion is that we shouldn't add code that is not necessary. With that being said, I won't block you on this if you insist.